### DISCUSSION TITLE
Struct Schema Adapter ?

### DISCUSSION THREAD

**Discussion by jbguerraz** [beltran/gohive#187] [Ideas]
Hello,

We're considering using gohive to register tables in a lake. Those tables host entities that are described using protobuf.
In order to create the table we need to describe its schema in a way that Hive understand.
Maybe we could contribute to this library so one could get a Hive schema out of a Golang structure: what do you think about it ? is it something of interest that would make sens to contribute?

Thank you!

**Comment by beltran** [comment]
Hello, I'm not sure if I understand, would this library take a protobuf file and generate the queries to create the database schema?

Thanks

**Comment by jbguerraz** [comment]
Hello @beltran ,
Sorry looks like I was not clear.
I meant to get the hive schema from the golang structure so it's generic.
